pkg/watch/manager.go

```diff
@@ -114,7 +86,8 @@ func (wm *Manager) updateManager() (bool, error) {
 	if err != nil {
 		return false, errp.Wrap(err, "error gathering watch changes, not restarting watch manager")
 	}
-	if wm.started && len(added) == 0 && len(removed) == 0 && len(changed) == 0 {
+	started := wm.started.Load().(bool)
```

Comment:
we should verify the type cast succeeded here

Reply:
Without verifying we panic. Normally this is bad, but in this case consistent with how Store() works:

https://golang.org/pkg/sync/atomic/#Value.Store

generics would be a better solution, but those don't exist.

b/c this error would happen early in execution, this is close to panicking on init(). If we don't panic, these errors would likely be swallowed.

Given the lack of ideal solution, does that change your opinion?

Reply:
> If we don't panic, these errors would likely be swallowed.

why is that? why not return the err and explicitly panic so it's cleaner?

Reply:
This relates to the comments on handling errors from `updateOrPause()` below.

Reply:
I was referring to cast issues in `wm.started.Load().(bool)` why is 
> If we don't panic, these errors would likely be swallowed.

Reply:
ok nvm i see it's called by the loop in start.

Reply:
cast, ok := var.(typecast) yields ok == false for a bad typecast

cast := var.(typecast) panics on failure.

https://golang.org/ref/spec#Type_assertions

Reply:
Or, if you're talking about why panic() at all? That's because this error gets swallowed per the referenced comment thread.